hardware/VirtualThermostat.h

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include <map>
+#include "ThermostatHardware.h"
+
+typedef std::map<int,  float  > T_Map_LastRoomTemp ;
+
+//duratoiopn of integration window in min
+#define INTEGRAL_DURATION 10
+
+// hors gel temperature celcius
+#define TEMPERATURE_HG   8
+// of temperature celcius
+#define TEMPERATURE_OFF  0
+
+enum VirtualThermostatMode {
+	Eco=0,
+	Confor,
+  FrostProtection,
+  Off,
+  EndMode
+};
+
+#define AVALAIBLE_MODE "Eco,Conf,Frost,Off"
+
+class CircularBuffer {
```

Comment:
I prefer if you define this class (and LastValue ) underneat the VirtualThermostat class

class VirtualThermostat()
{
private:
  class LastValue()
  {

Or use a namespace in the source/headerfile

Reason, Everywhere this class is included now, we can not use LastValue as it is defined here

Reply:
ok done